src/game/basketball/results.tsx

```diff
@@ -22,114 +23,232 @@ export function ResultComponent(props: {
+          simulationData[player.clientId]?.insideShotsMade,
+        simulationData[player.clientId]?.midShots -
+          simulationData[player.clientId]?.midShotsMade,
+        simulationData[player.clientId]?.outsideShots -
+          simulationData[player.clientId]?.outsideShotsMade,
+      ];
+      switch (index) {
+        case 0:
+          setPlayer1Data(playerMade);
+          setPlayer1MissedData(playerMissed);
+          break;
+        case 1:
+          setPlayer2Data(playerMade);
+          setPlayer2MissedData(playerMissed);
+          break;
+        case 2:
+          setPlayer3Data(playerMade);
+          setPlayer3MissedData(playerMissed);
+          break;
+        case 3:
+          setPlayer4Data(playerMade);
+          setPlayer4MissedData(playerMissed);
+          break;
+      }
+    }
+
+    setInsideScores(
+      controller.players.map(
+        (player) =>
+          (simulationData[player.clientId]?.insideShotsMade || 0) *
+          INSIDE_SHOT_POINTS_VALUE
+      )
+    );
+
+    setMidScores(
+      controller.players.map(
+        (player) =>
+          (simulationData[player.clientId]?.midShotsMade || 0) *
+          MID_SHOT_POINTS_VALUE
+      )
+    );
+
+    setOutsideScores(
+      controller.players.map(
+        (player) =>
+          (simulationData[player.clientId]?.outsideShotsMade || 0) *
+          OUTSIDE_SHOT_POINTS_VALUE
+      )
+    );
+
+    setPlayerLabels(controller.players.map((player) => player.name));
   }
 
   return (
-    <div>
-      <BarChart
-        xAxis={[
-          { scaleType: 'band', data: controller.players.map((p) => p.name) },
-        ]}
-        series={results}
-        slotProps={{ legend: { hidden: true } }}
-        width={window.innerWidth / 2 - 350}
-        height={window.innerHeight / 2 - 150}
-      />
-    </div>
+    <Stack sx={{ width: resultsWidth }} direction="column" alignItems="center">
+      <Stack direction="column" alignItems="center">
+        <Typography variant="h6">Score</Typography>
+        <BarChart
+          width={scoreChartWidth}
+          barLabel="value"
+          height={chartHeight}
+          series={[
+            {
+              data: insideScores,
+              label: 'In',
+              stack: 'mademissedshots',
+              color: '#e15759',
+            },
+            {
+              data: midScores,
+              label: 'Mid',
+              stack: 'mademissedshots',
+              color: '#ff9da7',
+            },
+            {
+              data: outsideScores,
+              label: '3pt',
+              stack: 'mademissedshots',
+              color: '#af7aa1',
+            },
+          ]}
+          xAxis={[{ data: playerLabels, scaleType: 'band' }]}
+        />
+      </Stack>
+      <Typography variant="h6">Shots</Typography>
+
+      <Stack direction="row" alignItems="center">
+        {controller.players.map((player, index) => (
+          <>
+            {index === 0 && (
```

Comment:
If I'm not mistaken, these Stack elements look like copies of eachother with just the player data changed, should then make a separate component that recieves the player data as a prop and re-use that

Reply:
I have separated it out into a function and passed data and some flag. Thinking that should be okay.